include/matx/transforms/cub.h

```diff
@@ -437,107 +437,40 @@ class matxCubPlan_t {
 #ifdef __CUDACC__
   if constexpr (is_tensor_view_v<InputOperator>)
   {
-
-    //////////////////////////////////////////////////////
-    //////           Rank 1 Tensors               ////////
-    //////////////////////////////////////////////////////
-    if (RANK == 1)
-    {
-      if (dir == SORT_DIR_ASC)
-      {
-        cub::DeviceSegmentedSort::SortKeys(
-            d_temp, temp_storage_bytes, a.Data(), a_out.Data(),
-            static_cast<int>(a.Size(RANK-1)),
-            1,
-            BeginOffset{a}, EndOffset{a},
-            stream);
-      }
-      else
-      {
-
-        cub::DeviceSegmentedSort::SortKeysDescending(
-            d_temp, temp_storage_bytes, a.Data(), a_out.Data(),
-            static_cast<int>(a.Size(RANK-1)),
-            1,
-            BeginOffset{a}, EndOffset{a},
-            stream);
-      }
+     // CUB added support for large items and segments in https://github.com/NVIDIA/cccl/pull/3308
+    int64_t _num_segments = 1;
+    for (int i = 0; i < RANK - 1; i++) {
```

Comment:
Should this be `RANK - 2`?

Reply:
I don't think so? We want `i = 0, ..., RANK - 2`

Reply:
Maybe I'm reading this wrong, but I would expect the number of segments to be everything but the last dimension's size. So if this is looping from 0..RANK-1, it's looping over all sizes of the operator. So this is really `_num_items`?

Reply:
the condition is `< RANK - 1`, so `i=RANK-1` won't be included in the loop.